OpenTrons/opentrons-api
# feature(robot-server, api): get a runs loaded labware definitions

# Overview

add's GET `/runs/{runId}/loaded_labware_definitions` route to robot-server.
closes https://opentrons.atlassian.net/browse/RSS-240.
work towards removing `LabwareDefinition` from `loadLabware` and `loadModule` response. 


# Test Plan

- Tested on local server and with an end to end test. works as expected.
- Need to test via app.
- upload a protocol with loadLabware commands with custom and common labware. 
- start the run.
- Make sure hitting GET `/runs/{runId}/loaded_labware_definitions` returns the loaded labware list.


# Changelog

- Added GET route to the labware_router to get the run's loaded labware definition.
- Added a method in the `LabwareView` and In the `RunDataManager` to fetch run's loaded labware definition.
- Added a response type for lists. 


# Review requests

does the changes make sense? did I miss anything? 
raised exception makes sense? we are not persisting the labware definition in the db so we will only return the labware definition for the current run. we are persisting commands. can strip it from the loadLabware commands but I don't think this is necessary? @b-cooper . 

# Risk assessment

low. added another route should not effect existing logic.
